Source/WebCore/rendering/RenderHTMLCanvas.cpp

```diff
@@ -71,6 +71,18 @@ bool RenderHTMLCanvas::requiresLayer() const
     return canvasCompositingStrategy(*this) != CanvasPaintedToEnclosingLayer;
 }
 
+void RenderHTMLCanvas::styleDidChange(StyleDifference diff, const RenderStyle* oldStyle)
```

Comment:
I'm not sure this is right; a style change doesn't mean that the context state changed, and calling `contentChanged` on every style change seems wrong.

Why doesn't the call to `canvasRenderer->contentChanged(ContentChangeType::Canvas)`  in `HTMLCanvasElement::reset()` already take care of this?

Reply:
You right. I removed this code and I used `RenderLayerBacking::updateDrawsContent()` to check whether the renderer of `RenderLayer` itself drawsHDRContent or not and `GraphicsLayer::setDrawsHDRContent()` accordingly.